addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/OceanicThingHandler.java

```diff
@@ -231,14 +77,49 @@ public void run() {
                     }
                 }
             } catch (Exception e) {
-                logger.error("An exception occurred while polling the Oceanic Water Softener: '{}'", e.getMessage());
+                logger.error("An exception occurred while polling the Oceanic Water Softener: '{}'", e.getMessage(), e);
+                scheduler.schedule(resetRunnable, 0, TimeUnit.SECONDS);
             }
         }
     };
 
+    private Runnable resetRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            dispose();
+            initialize();
+        }
+    };
+
     @Override
-    public void handleCommand(ChannelUID channelUID, Command command) {
+    public void initialize() {
+        if (getConfig().get(BUFFER_SIZE) == null) {
+            bufferSize = 1024;
+        } else {
+            bufferSize = (int) getConfig().get(BUFFER_SIZE);
```

Comment:
Afaik the confit will never return something that can be caster to an int?

Reply:
true. it is an artefact from the past :-)